middleware/grpc_logging.go

```diff
@@ -14,10 +16,34 @@ const gRPC = "gRPC"
 var ServerLoggingInterceptor = func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
 	begin := time.Now()
 	resp, err := handler(ctx, req)
+	entry := LogEntryFromContext(ctx).WithFields(log.Fields{"method": info.FullMethod, "duration": time.Since(begin)})
 	if err != nil {
-		log.Warnf("%s %s (%v) %s", gRPC, info.FullMethod, err, time.Since(begin))
+		entry.WithError(err).Warn(gRPC)
 	} else {
-		log.Debugf("%s %s (success) %s", gRPC, info.FullMethod, time.Since(begin))
+		entry.Debugf("%s (success)", gRPC)
 	}
 	return resp, err
 }
+
+// LogEntryFromContext returns a logrus Entry that has information
+// about the current user in its details.
+//
+// e.g.
+//   entry := middleware.LogEntryFromContext(ctx)
+//   entry.Errorf("Could not chunk chunks: %v", err)
+func LogEntryFromContext(ctx context.Context) *log.Entry {
+	// Weaveworks uses "orgs" and "orgID" to represent Cortex users,
+	// even though the code-base generally uses `userID` to refer to the same thing.
+	userID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return log.NewEntry(log.StandardLogger())
+	}
+	return LogEntryFromUserID(userID)
+}
+
+// LogEntryFromUserID returns a log Entry that has information about
+// the current user in its details.
+func LogEntryFromUserID(userID string) *log.Entry {
+	// See note in WithContext.
+	return log.NewEntry(log.StandardLogger()).WithField("orgID", userID)
+}
```

Comment:
> I don't think removing those comments will help the confusion much, overall.

Removing the comments alone wouldn't. But inlining and removing them would since then the code straightforwardly extracts the org id and logs it. No mention of any user id.